core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -64,4 +77,21 @@ public static boolean hasDeletes(FileScanTask task) {
             splitFiles),
         BaseCombinedScanTask::new);
   }
+
+  public static Map<StructLikeWrapper, Collection<FileScanTask>> groupTasksByPartition(
+      PartitionSpec spec,
```

Comment:
I don't think it's correct to use the table latest partition spec to group the `FileScanTask`,  because different `FileScanTask`  many have different partition specs,  the correct way is to use the `FileScanTask#spec` to group the tasks.   We should remove the `spec` as an argument, otherwise it introducing a bug...